File: javascript/1408.String-Matching-in-an-Array.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function (words) {
  words.sort((a, b) => a.length - b.length);
  const result = new Set();

  for (let i = 0; i < words.length; i++) {
    for (let j = i + 1; j < words.length; j++) {
      if (words[j].includes(words[i])) {
        result.add(words[i]);
      }
    }
  }

  return Array.from(result);
};

File: python/1408.String-Matching-in-an-Array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        words.sort(key=len)
        result = set()

        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if words[i] in words[j]:
                    result.add(words[i])

        return list(result)